File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/TypeTempalteService.java
package com.pinyougou.sellergoods.service;


import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbTypeTemplate;
import com.pinyougou.query.TypeTempalteQuery;

import entity.PageResult;
/**
 * 数据模板业务接口类
 * @author 89568
 *
 */
public interface TypeTempalteService {
	/**
	 * 查询条件显示
	 * @param query
	 * @param num
	 * @param size
	 * @return
	 */
	public PageResult pageList(TypeTempalteQuery query,int num,int size);
	/**
	 * 新增模板
	 * @param typeTemplate
	 */
	public void seavOfupdata(TbTypeTemplate typeTemplate);
	/**
	 * 根据主键id查找
	 * @param id
	 * @return
	 */
	public TbTypeTemplate findOne(Long id);
	/**
	 * 批量删除id
	 * @param id
	 */
	public void deleteById(Long[] id);
	
	/**
	 * 返回规格列表
	 * @return
	 */
	public List<Map> findSpecList(Long id);

}

File: pinyougou-shop-web/src/main/webapp/js/service/TypeTempalteService.js

//service TypeTempalte
app.service("TypeTempalteService",function($http){
	//分页查询条件 显示所有
	this.findPage = function(num,size,entity){
		
		return $http.post("../TypeTempalte/findPage.do?num="+num+"&size="+size,entity);
	}
	//新增模板或者修改模板
	this.sevaOfupdata = function(entity){
		
		return $http.post("../TypeTempalte/seavOfupdata.do",entity);
	}
	//查询id模板显示可修改
	this.findOne = function(id){
		return $http.post("../TypeTempalte/findId.do?id="+id);
	}
	//删除模板
	this.deleteById=function(id){
		return $http.get("../TypeTempalte/deleteType.do?id="+id);
	}
	
	//回显JSON specids
	this.findSpecList = function (id){
		return $http.get("../TypeTempalte/findSpecList.do?id="+id);
	}
})